[32mINFO    [0m pytest_operator.plugin:plugin.py:937 Model status:

Model    Controller          Cloud/Region        Version  SLA          Timestamp
testing  concierge-microk8s  microk8s/localhost  3.6.7    unsupported  01:55:02Z

App                       Version  Status  Scale  Charm                     Channel        Rev  Address         Exposed  Message
pgbouncer-k8s             1.21.0   active      2  pgbouncer-k8s                              0  10.152.183.112  no       
postgresql-k8s            14.18    active      3  postgresql-k8s            14/edge        610  10.152.183.163  no       
postgresql-test-app                active      1  postgresql-test-app       latest/edge    383  10.152.183.254  no       
self-signed-certificates           active      1  self-signed-certificates  latest/stable  264  10.152.183.56   no       

Unit                         Workload  Agent  Address      Ports  Message
pgbouncer-k8s/0              active    idle   10.1.36.201         
pgbouncer-k8s/1*             active    idle   10.1.36.202         
postgresql-k8s/0             active    idle   10.1.36.208         
postgresql-k8s/1*            active    idle   10.1.36.210         Primary
postgresql-k8s/2             active    idle   10.1.36.209         
postgresql-test-app/0*       active    idle   10.1.36.203         
self-signed-certificates/0*  active    idle   10.1.36.207         

[32mINFO    [0m pytest_operator.plugin:plugin.py:943 Juju error logs:

unit-pgbouncer-k8s-1: 01:47:26 ERROR unit.pgbouncer-k8s/1.juju-log backend-database:7: deferring database-created hook - postgres database not ready
unit-postgresql-k8s-0: 01:47:26 ERROR unit.postgresql-k8s/0.juju-log pgdata folder not found in /var/lib/postgresql/data/pgdata
unit-postgresql-k8s-2: 01:47:27 ERROR unit.postgresql-k8s/2.juju-log pgdata folder not found in /var/lib/postgresql/data/pgdata
unit-pgbouncer-k8s-1: 01:47:29 ERROR unit.pgbouncer-k8s/1.juju-log deferring database-created hook - postgres database not ready
unit-pgbouncer-k8s-1: 01:47:38 ERROR unit.pgbouncer-k8s/1.juju-log deferring database-created hook - postgres database not ready
unit-pgbouncer-k8s-1: 01:47:47 ERROR unit.pgbouncer-k8s/1.juju-log deferring database-created hook - postgres database not ready
unit-pgbouncer-k8s-1: 01:47:58 ERROR unit.pgbouncer-k8s/1.juju-log deferring database-created hook - postgres database not ready
unit-postgresql-k8s-2: 01:48:02 ERROR unit.postgresql-k8s/2.juju-log pgdata folder not found in /var/lib/postgresql/data/pgdata
unit-postgresql-k8s-0: 01:48:02 ERROR unit.postgresql-k8s/0.juju-log pgdata folder not found in /var/lib/postgresql/data/pgdata
unit-pgbouncer-k8s-1: 01:48:06 ERROR unit.pgbouncer-k8s/1.juju-log deferring database-created hook - postgres database not ready
unit-pgbouncer-k8s-1: 01:48:15 ERROR unit.pgbouncer-k8s/1.juju-log deferring database-created hook - postgres database not ready
unit-pgbouncer-k8s-1: 01:48:24 ERROR unit.pgbouncer-k8s/1.juju-log deferring database-created hook - postgres database not ready
unit-postgresql-k8s-0: 01:48:33 ERROR unit.postgresql-k8s/0.juju-log pgdata folder not found in /var/lib/postgresql/data/pgdata
unit-postgresql-k8s-2: 01:48:33 ERROR unit.postgresql-k8s/2.juju-log pgdata folder not found in /var/lib/postgresql/data/pgdata
unit-pgbouncer-k8s-1: 01:48:43 ERROR unit.pgbouncer-k8s/1.juju-log pgb-peers:0: 
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-pgbouncer-k8s-1/charm/lib/charms/postgresql_k8s/v0/postgresql.py", line 442, in enable_disable_extensions
    with self._connect_to_database(
  File "/var/lib/juju/agents/unit-pgbouncer-k8s-1/charm/lib/charms/postgresql_k8s/v0/postgresql.py", line 180, in _connect_to_database
    connection = psycopg2.connect(
  File "/var/lib/juju/agents/unit-pgbouncer-k8s-1/charm/venv/lib/python3.10/site-packages/psycopg2/__init__.py", line 122, in connect
    conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
psycopg2.OperationalError: connection to server at "postgresql-k8s-primary.testing.svc.cluster.local" (10.152.183.106), port 5432 failed: FATAL:  no pg_hba.conf entry for host "10.1.36.202", user "relation_id_7", database "postgresql_test_app_database", no encryption


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-pgbouncer-k8s-1/charm/src/relations/pgbouncer_provider.py", line 162, in _on_database_requested
    self.charm.backend.postgres.create_database(
  File "/var/lib/juju/agents/unit-pgbouncer-k8s-1/charm/lib/charms/postgresql_k8s/v0/postgresql.py", line 261, in create_database
    self.enable_disable_extensions(dict.fromkeys(plugins, True), database)
  File "/var/lib/juju/agents/unit-pgbouncer-k8s-1/charm/lib/charms/postgresql_k8s/v0/postgresql.py", line 457, in enable_disable_extensions
    raise PostgreSQLEnableDisableExtensionError() from e
charms.postgresql_k8s.v0.postgresql.PostgreSQLEnableDisableExtensionError
unit-postgresql-k8s-2: 01:49:12 ERROR unit.postgresql-k8s/2.juju-log pgdata folder not found in /var/lib/postgresql/data/pgdata
unit-postgresql-k8s-0: 01:49:12 ERROR unit.postgresql-k8s/0.juju-log pgdata folder not found in /var/lib/postgresql/data/pgdata
unit-postgresql-k8s-0: 01:49:35 ERROR unit.postgresql-k8s/0.juju-log pgdata folder not found in /var/lib/postgresql/data/pgdata
unit-postgresql-k8s-2: 01:49:35 ERROR unit.postgresql-k8s/2.juju-log pgdata folder not found in /var/lib/postgresql/data/pgdata
unit-postgresql-k8s-0: 01:49:43 ERROR unit.postgresql-k8s/0.juju-log pgdata folder not found in /var/lib/postgresql/data/pgdata
unit-postgresql-k8s-2: 01:49:43 ERROR unit.postgresql-k8s/2.juju-log pgdata folder not found in /var/lib/postgresql/data/pgdata
unit-postgresql-k8s-2: 01:50:52 ERROR unit.postgresql-k8s/2.juju-log pgdata folder not found in /var/lib/postgresql/data/pgdata
unit-postgresql-k8s-2: 01:51:01 ERROR unit.postgresql-k8s/2.juju-log pgdata folder not found in /var/lib/postgresql/data/pgdata

[32mINFO    [0m pytest_operator.plugin:plugin.py:1025 Forgetting model main...